drivers/i2c/Kconfig.nrfx

```diff
@@ -17,7 +17,9 @@ menuconfig I2C_NRFX
 if I2C_NRFX
 
 # Nordic TWIx0 and SPIx0 instances can not be used at the same time.
-if I2C_0 && !SPI_0
+# nRF52810 is an exception to this rule, since in this SoC a bit different
```

Comment:
I would recommend to copy the exact same text you added in the PR description, as that one better describes the reason for this change.